File: Periodic Practice1/shape.py
n, k = map(int, input().split())

pattern = []

for i in range(n):
    pattern.append(input())


def solve(x, y, k):

    if k == 0:
        return "."

    blocksize = pow(n, k - 1)

    row = x // blocksize
    col = y // blocksize

    if pattern[row][col] == "*":
        return "*"

    x %= blocksize
    y %= blocksize

    return solve(x, y, k - 1)


size = pow(n, k)

for i in range(size):
    for j in range(size):
        print(solve(i, j, k), end="")
    print()

File: Periodic Practice1/shape.cpp
#include <bits/stdc++.h>
using namespace std;

int n, k;
vector<string> pattern;

char solve(int x, int y, int level) {

    if (level == 0) {
        return '.';
    }

    int blockSize = pow(n, level - 1);

    int row = x / blockSize;
    int col = y / blockSize;

    if (pattern[row][col] == '*') {
        return '*';
    }

    x %= blockSize;
    y %= blockSize;

    return solve(x, y, level - 1);
}

int main() {

    cin >> n >> k;

    pattern.resize(n);

    for (int i = 0; i < n; i++) {
        cin >> pattern[i];
    }

    int size = pow(n, k);

    for (int i = 0; i < size; i++) {
        for (int j = 0; j < size; j++) {
            cout << solve(i, j, k);
        }
        cout << endl;
    }
}